pkg/gadgets/trace/exec/tracer/tracer.go

```diff
@@ -134,6 +143,7 @@ func (t *Tracer) run() {
 			continue
 		}
 
+		// this works regardless the kind of event because cwd is defined at the end of the structure
 		bpfEvent := (*execsnoopEvent)(unsafe.Pointer(&record.RawSample[0]))
```

Comment:
Args has to be the last field because the ebpf program has some optimization not to send the full event but only the bytes that are actually needed. I added a comment clarifying that cwd is at the end but before args.